src/plugins/vis_augmenter/server/plugin.ts

```diff
@@ -23,42 +26,53 @@ export interface VisAugmenterPluginStart {}
 export class VisAugmenterPlugin
   implements Plugin<VisAugmenterPluginSetup, VisAugmenterPluginStart> {
   private readonly logger: Logger;
+  private readonly config$: Observable<VisAugmenterPluginConfigType>;
 
   constructor(initializerContext: PluginInitializerContext) {
     this.logger = initializerContext.logger.get();
+    this.config$ = initializerContext.config.create<VisAugmenterPluginConfigType>();
   }
 
-  public setup(core: CoreSetup) {
+  public async setup(core: CoreSetup) {
     this.logger.debug('VisAugmenter: Setup');
     core.savedObjects.registerType(augmentVisSavedObjectType);
     core.capabilities.registerProvider(capabilitiesProvider);
 
-    core.uiSettings.register({
-      ['visualization:enablePluginAugmentation']: {
-        name: i18n.translate('visualization.enablePluginAugmentationTitle', {
-          defaultMessage: 'Enable plugin augmentation',
-        }),
-        value: true,
-        description: i18n.translate('visualization.enablePluginAugmentationText', {
-          defaultMessage: 'Plugin functionality can be accessed from line chart visualizations',
-        }),
-        category: ['visualization'],
-        schema: schema.boolean(),
-      },
-      ['visualization:enablePluginAugmentation.maxPluginObjects']: {
-        name: i18n.translate('visualization.enablePluginAugmentation.maxPluginObjectsTitle', {
-          defaultMessage: 'Max number of associated augmentations',
-        }),
-        value: 10,
-        description: i18n.translate('visualization.enablePluginAugmentation.maxPluginObjectsText', {
-          defaultMessage:
-            'Associating more than 10 plugin resources per visualization can lead to performance ' +
-            'issues and increase the cost of running clusters.',
-        }),
-        category: ['visualization'],
-        schema: schema.number({ min: 0 }),
-      },
-    });
+    const config: VisAugmenterPluginConfigType = await this.config$.pipe(first()).toPromise();
+    const isAugmentationEnabled =
+      config.pluginAugmentationEnabled === undefined ? true : config.pluginAugmentationEnabled;
+
+    if (isAugmentationEnabled) {
+      core.uiSettings.register({
+        ['visualization:enablePluginAugmentation']: {
+          name: i18n.translate('visualization.enablePluginAugmentationTitle', {
+            defaultMessage: 'Enable plugin augmentation',
+          }),
+          value: true,
+          description: i18n.translate('visualization.enablePluginAugmentationText', {
+            defaultMessage: 'Plugin functionality can be accessed from line chart visualizations',
+          }),
+          category: ['visualization'],
+          schema: schema.boolean(),
+        },
+        ['visualization:enablePluginAugmentation.maxPluginObjects']: {
+          name: i18n.translate('visualization.enablePluginAugmentation.maxPluginObjectsTitle', {
+            defaultMessage: 'Max number of associated augmentations',
```

Comment:
This message seems a little weird, should it be something like 'max number of associated plugin objects / plugin resources'? Not sure if this was finalized by tech writer

Reply:
This was based on mockups and we can improve this.

Reply:
Sure - no blocker here. We can probably follow up with one larger PR to update wording after UX has a full walk through.